osutil/epoll/epoll.go

```diff
@@ -203,7 +228,32 @@ func (e *Epoll) WaitTimeout(duration time.Duration) ([]Event, error) {
 		}
 		events = append(events, event)
 	}
-	return events, nil
+	eventCh <- events
+	return
+}
+
+// WaitTimeout blocks and waits with the given timeout for arrival of events on any of the added file descriptors.
+//
+// A duration value of -1 milliseconds disables timeout.
+//
+// Please refer to epoll_wait(2) and EPOLL_WAIT for details.
+//
+// Warning, using epoll from Golang explicitly is tricky.
+func (e *Epoll) WaitTimeout(duration time.Duration) ([]Event, error) {
+	if e.fd == -1 {
+		return nil, ErrEpollClosed
+	}
+	eventCh := make(chan []Event, 1)
+	errCh := make(chan error, 1)
+	go e.waitTimeoutInternal(duration, eventCh, errCh)
```

Comment:
`e.Close` may write to `e.fd` at the same time that the goroutine running `e.waitTimeoutInternal` is reading from `e.fd`